packages/venia-concept/src/components/Checkout/paymentsForm.js

```diff
@@ -22,228 +21,232 @@ const DEFAULT_FORM_VALUES = {
-            street0: string,
-            textInput: string
-        }),
-        initialValues: shape({
-            city: string,
-            postcode: string,
-            region_code: string,
-            sameAsShippingAddress: bool,
-            street0: array
-        }),
-        submit: func.isRequired,
-        submitting: bool,
-        countries: array
-    };
-
-    static defaultProps = {
-        initialValues: {}
-    };
-
-    state = {
-        isRequestingPaymentNonce: false
-    };
-
-    render() {
-        const { classes, initialValues } = this.props;
-        const { formChildren } = this;
+const BillingAddressFields = ({ classes, countries }) => {
+    return (
+        <Fragment>
+            <div className={classes.street0}>
+                <Field label="Street">
+                    <TextInput
+                        id={classes.street0}
+                        field="street[0]"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+            <div className={classes.city}>
+                <Field label="City">
+                    <TextInput
+                        id={classes.city}
+                        field="city"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+            <div className={classes.region_code}>
+                <Field label="State">
+                    <TextInput
+                        id={classes.region_code}
+                        field="region_code"
+                        validate={combine([
+                            isRequired,
+                            [hasLengthExactly, 2],
+                            [validateRegionCode, countries]
+                        ])}
+                    />
+                </Field>
+            </div>
+            <div className={classes.postcode}>
+                <Field label="ZIP">
+                    <TextInput
+                        id={classes.postcode}
+                        field="postcode"
+                        validate={isRequired}
+                    />
+                </Field>
+            </div>
+        </Fragment>
+    );
+};
 
-        let initialFormValues;
-        if (isObjectEmpty(initialValues)) {
-            initialFormValues = DEFAULT_FORM_VALUES;
-        }
-        // We have some initial values, use them.
-        else {
-            if (initialValues.sameAsShippingAddress) {
-                // If the addresses are the same, don't populate any fields
-                // other than the checkbox with an initial value.
-                initialFormValues = {
-                    addresses_same: true
+const FormChildren = props => {
+    const {
+        cancel,
+        classes,
+        countries,
+        formApi,
+        isRequestingPaymentNonce,
+        setIsRequestingPaymentNonce,
+        submit,
+        submitting
+    } = props;
+    const formState = useFormState();
```

Comment:
A wild hook appears!

Reply:
Good comment. I'm not surprised, though; Informed implements change publishing manually, so they're not doing hooks + context in any sort of optimal way.